app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -381,13 +381,7 @@ def get_content_view_environments(label = nil, organization = nil)
           where("#{Organization.table_name}.id = ?", organization.id)
       environments = environments.where("#{Katello::ContentViewEnvironment.table_name}.label = ?", label) if label
 
-      environments.delete_if do |env|
-        if env.content_view.default
-          !env.environment.readable?
-        else
-          !env.content_view.readable?
-        end
-      end
+      environments = environments.to_a.delete_if { |env| (env.content_view.default && !env.environment.readable?) || !env.content_view.readable? }
```

Comment:
@johnpmitsch I was slightly mistaken. delete_if will return the modified collection but it is not the same object as the original collection. If you use the delete_if statement as your last line, you'll be good because of ruby's implicit return. If you state environments again after the delete_if you'll return the original collection (not what you intended). So in that case you would need to do the variable assignment.